common/djangoapps/course_modes/views.py

```diff
@@ -226,3 +228,31 @@ def _get_requested_mode(self, request_dict):
             return 'honor'
         else:
             return None
+
+
+@ensure_csrf_cookie
+@require_level('staff')
+def add_mode_to_course(request, course_id):
+    """
+    Create the honor mode for the course_id
+    """
+    if request.method == 'GET':
+        return render_to_response("course_modes/add_course_mode.html", {'course_id': course_id})
+    if request.method == 'POST':
+        course_mode = request.POST.get('course_mode')
+        course_mode_display_name = request.POST.get('course_mode_display_name')
+        try:
+            course_mode_price = int(request.POST.get('course_mode_price'))
+        except ValueError:
+            return HttpResponseBadRequest('Enter the integer value for the price')
+        course_mode_currency = request.POST.get('course_mode_currency')
+
+        course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+        CourseMode.create_course_mode(
```

Comment:
If exposing create_course_mode() from an HTTP endpoint, we'll definitely need to validate the types of modes being created. Two things I can think of off the top of my head are the acceptable mode types, and we only currently support having one "paid" course mode. That'd be a check on the minimum price